artichoke-backend/src/extn/core/time/args.rs

```diff
@@ -20,33 +21,140 @@ impl Default for TimeArgs {
             hour: 0,
             minute: 0,
             second: 0,
-            nanoseconds: 0,
+            micros: 0,
         }
     }
 }
 
-pub fn as_time_args<T>(interp: &mut Artichoke, args: T) -> Result<TimeArgs, Error>
-where
-    T: IntoIterator<Item = Value>,
-{
-    let _interp = interp;
-    let _args = args;
-    Ok(TimeArgs::default())
+impl TimeArgs {
+    pub fn year(&self) -> Result<i32, Error> {
+        i32::try_from(self.year).map_err(|_| ArgumentError::with_message("year out of range").into())
+    }
+
+    pub fn month(&self) -> Result<u8, Error> {
+        // 1..=12 is gauranteed to fit into u8, so this cast is safe
```

Comment:
typo: gauranteed -> guaranteed

Reply:
`Abolish gaurantee{,s,d} guarantee{}` to the rescue!